epochStart/bootstrap/storageProcess.go

```diff
@@ -0,0 +1,501 @@
+package bootstrap
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/core/partitioning"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/endProcess"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	factoryDataPool "github.com/ElrondNetwork/elrond-go/dataRetriever/factory"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever/factory/containers"
+	storageResolversContainers "github.com/ElrondNetwork/elrond-go/dataRetriever/factory/storageResolversContainer"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever/requestHandlers"
+	"github.com/ElrondNetwork/elrond-go/epochStart"
+	"github.com/ElrondNetwork/elrond-go/epochStart/notifier"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	storageFactory "github.com/ElrondNetwork/elrond-go/storage/factory"
+	"github.com/ElrondNetwork/elrond-go/storage/timecache"
+)
+
+// ArgsStorageEpochStartBootstrap holds the arguments needed for creating an epoch start data provider component
+// from storage
+type ArgsStorageEpochStartBootstrap struct {
+	ArgsEpochStartBootstrap
+	ImportDbConfig      config.ImportDbConfig
+	ChanGracefullyClose chan endProcess.ArgEndProcess
+}
+
+type storageEpochStartBootstrap struct {
+	*epochStartBootstrap
+	resolvers           dataRetriever.ResolversContainer
+	store               dataRetriever.StorageService
+	importDbConfig      config.ImportDbConfig
+	chanGracefullyClose chan endProcess.ArgEndProcess
+	chainID             string
+}
+
+// NewStorageEpochStartBootstrap will return a new instance of storageEpochStartBootstrap that can bootstrap
+// the node with the help of storage resolvers through the import-db process
+func NewStorageEpochStartBootstrap(args ArgsStorageEpochStartBootstrap) (*storageEpochStartBootstrap, error) {
+	esb, err := NewEpochStartBootstrap(args.ArgsEpochStartBootstrap)
+	if err != nil {
+		return nil, err
+	}
+
+	if args.ChanGracefullyClose == nil {
+		return nil, dataRetriever.ErrNilGracefullyCloseChannel
+	}
+
+	sesb := &storageEpochStartBootstrap{
+		epochStartBootstrap: esb,
+		importDbConfig:      args.ImportDbConfig,
+		chanGracefullyClose: args.ChanGracefullyClose,
+		chainID:             args.CoreComponentsHolder.ChainID(),
+	}
+
+	return sesb, nil
+}
+
+// Bootstrap runs the fast bootstrap method from local storage or from import-db directory
+func (sesb *storageEpochStartBootstrap) Bootstrap() (Parameters, error) {
+	if !sesb.generalConfig.GeneralSettings.StartInEpochEnabled {
+		log.Warn("fast bootstrap is disabled")
+
+		sesb.initializeFromLocalStorage()
+		if !sesb.baseData.storageExists {
+			err := sesb.createTriesComponentsForShardId(sesb.genesisShardCoordinator.SelfId())
+			if err != nil {
+				return Parameters{}, err
+			}
+
+			return Parameters{
+				Epoch:       sesb.startEpoch,
```

Comment:
it seems that the startEpoch is set only if HardFork. Wouldn't a startEpoch also for import-db help decrease the computation time by starting at a higher epoch?